doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,889 @@
+examples.
+
+```c++ (C++ish pseudocode)
+     template<typename TOutput> TOutput ActivateByCoCreate(REFCLSID rclsid) {
+        winrt::com_ptr<::IInspectable> result;
+        check_hresult(::CoCreateInstance(rclsid, nullptr, CLSCTX_LOCAL_SERVER, IID_PPV_ARGS(&result)));
+        return result.as<TOutput>();
+    }
+
+    AppInstaller CreateAppInstaller() {
+        return ActivateByCoCreate<AppInstaller>(CLSID_AppInstaller);
+    }
+    InstallOptions CreateInstallOptions() {
+        return ActivateByCoCreate<InstallOptions>(CLSID_InstallOptions);
+    }
+    FindPackagesOptions CreateFindPackagesOptions() {
+        return ActivateByCoCreate<FindPackagesOptions>(CLSID_FindPackagesOptions);
+    }
+    GetCompositeAppCatalogOptions CreateGetCompositeAppCatalogOptions() {
+        return ActivateByCoCreate<GetCompositeAppCatalogOptions>(CLSID_GetCompositeAppCatalogOptions);
+    }
+    PackageMatchFilter CreatePackageMatchFilter() {
+        return ActivateByCoCreate<PackageMatchFilter>(CLSID_PackageMatchFilter);
+    }
+```
+
+## 3.1. Search
+
+The api can be used to search for packages in a catalog known to Windows Package Manager. This can
+be used to get availability information or start an install.
+
+```c++ (C++ish pseudocode)
+
+    IAsyncOperation<CatalogPackage> FindPackageInCatalog(AppCatalog catalog, std::wstring packageId)
+    {
+        FindPackagesOptions findPackagesOptions = CreateFindPackagesOptions();
+        PackageMatchFilter filter = CreatePackageMatchFilter();
+        filter.IsAdditive(true);
+        filter.Field(PackageMatchField::Id);
+        filter.Type(MatchType::Exact);
+        filter.Value(packageId);
+        findPackagesOptions.Filters().Append(filter);
+        FindPackagesResult findPackagesResult{ co_await catalog.FindPackagesAsync(findPackagesOptions) };
+
+        winrt::IVectorView<ResultMatch> matches = findPackagesResult.Matches();
+        co_return matches.GetAt(0).CatalogPackage();
+    }
+
+    IAsyncOperation<CatalogPackage> MainPage::FindPackage()
+    {
+        // Capture the ui thread context.
+        co_await winrt::resume_background();
+
+        AppInstaller appInstaller = CreateAppInstaller();
+        AppCatalog catalog{ appInstaller.GetAppCatalog(PredefinedAppCatalog::OpenWindowsCatalog) };
+        catalog.OpenAsync().get();
+        co_return FindPackageInCatalog(catalog, m_installAppId).get();
+    }
+```
+
+## 3.2. Install
+
+```c++ (C++ish pseudocode)
+
+    IAsyncOperationWithProgress<InstallResult, InstallProgress> InstallPackage(CatalogPackage package)
+    {
+        AppInstaller appInstaller = CreateAppInstaller();
+        InstallOptions installOptions = CreateInstallOptions();
+
+        installOptions.AppInstallScope(AppInstallScope::User);
+        installOptions.CatalogPackage(package);
+        installOptions.AppInstallMode(AppInstallMode::Silent);
+
+        return appInstaller.InstallPackageAsync(installOptions);

```

Comment:
Can I pass `null` as the options to get "default options"?

Reply:
No, since package isn't optional